File: src/main/java/me/lordsaad/modeoff/api/PlotChunkCache.java
package me.lordsaad.modeoff.api;

import net.minecraft.block.state.IBlockState;
import net.minecraft.init.Blocks;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.ChunkCache;
import net.minecraft.world.World;

/**
 * Created by Gegy
 */
public class PlotChunkCache extends ChunkCache {
	private final BlockPos min;
	private final BlockPos max;

	public PlotChunkCache(World world, BlockPos from, BlockPos to, int subIn) {
		super(world, from, to, subIn);
		min = new BlockPos(Math.min(from.getX(), to.getX()), Math.min(from.getY(), to.getY()), Math.min(from.getZ(), to.getZ()));
		max = new BlockPos(Math.max(from.getX(), to.getX()), Math.max(from.getY(), to.getY()), Math.max(from.getZ(), to.getZ()));
	}

	@Override
	public IBlockState getBlockState(BlockPos pos) {
		if (pos.getX() >= min.getX() && pos.getY() >= min.getY() && pos.getZ() >= min.getZ() && pos.getX() < max.getX() && pos.getY() < max.getY() && pos.getZ() < max.getZ()) {
			return super.getBlockState(pos);
		}
		return Blocks.AIR.getDefaultState();
	}
}

File: src/main/kotlin/info/modoff/modeoff/client/plot/PlotChunkCache.kt
package info.modoff.modeoff.client.plot

import net.minecraft.block.state.IBlockState
import net.minecraft.init.Blocks
import net.minecraft.util.math.BlockPos
import net.minecraft.world.ChunkCache
import net.minecraft.world.World

/**
 * Created by Gegy
 */
class PlotChunkCache(world: World, from: BlockPos, to: BlockPos, subIn: Int) : ChunkCache(world, from, to, subIn) {
    private val min: BlockPos = BlockPos(Math.min(from.x, to.x), Math.min(from.y, to.y), Math.min(from.z, to.z))
    private val max: BlockPos = BlockPos(Math.max(from.x, to.x), Math.max(from.y, to.y), Math.max(from.z, to.z))

    override fun getBlockState(pos: BlockPos): IBlockState {
        if (pos.x >= min.x && pos.y >= min.y && pos.z >= min.z && pos.x < max.x && pos.y < max.y && pos.z < max.z) {
            return super.getBlockState(pos)
        }
        return Blocks.AIR.defaultState
    }
}
